spec/notifier_spec.rb

```diff
@@ -522,6 +522,12 @@ def to_json(*)
     end
 
     context "given a non-exception with calculated internal frames only" do
+      it "prevents mutation of passed-in params hash" do
```

Comment:
I don't think this test corresponds to the description. You check for the presence of `dup`, not if the object was mutated. Mutated objects carry the same object id. Thus, this test should be changed to something more sturdy:

```rb
params = { only_this_item: true }
notice = @airbrake.build_notice(RuntimeError.new('bingo'), params)
notice[:params][:extra_item] = true
expect(params).not_to include(:extra_item)
```